File: event-service/src/main/java/org/tggc/eventservice/repository/ParticipantRepository.java
package org.tggc.eventservice.repository;

import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;
import org.tggc.eventservice.model.Participant;

import java.util.Optional;

@Repository
public interface ParticipantRepository extends JpaRepository<Participant, Long> {

    Optional<Participant> findByEventId(Long eventId);
}

File: event-service/src/main/kotlin/org/tggc/eventservice/repository/ParticipantRepository.kt
package org.tggc.eventservice.repository

import org.springframework.data.r2dbc.repository.R2dbcRepository
import org.springframework.stereotype.Repository
import org.tggc.eventservice.model.Participant
import reactor.core.publisher.Flux
import reactor.core.publisher.Mono

@Repository
interface ParticipantRepository : R2dbcRepository<Participant, Long> {

    fun findByEventId(eventId: Long?): Flux<Participant>

    fun existsByEventIdAndUserId(eventId: Long, userId: Long): Mono<Boolean>

    fun deleteByEventId(eventId: Long): Mono<Void>
}
